core-reactive/src/main/java/io/micronaut/core/async/flow/ReactiveFlowImpl.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.async.flow;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.flow.CompletableFutureFlow;
+import io.micronaut.core.flow.Flow;
+import io.micronaut.core.flow.ImperativeFlow;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * The reactive flow implementation.
+ *
+ * @author Denis Stepanov
+ * @since 4.0.0
+ */
+@Internal
+final class ReactiveFlowImpl implements ReactiveFlow<Object> {
```

Comment:
should be `ReactorFlow` or something since this is for reactor?

Reply:
Renamed